src/pages/global-authorization/index.js

```diff
@@ -0,0 +1,200 @@
+import { Box, Button, Grid, Tooltip, Typography, IconButton } from '@mui/material'
+import Icon from 'src/@core/components/icon'
+import { useContext, useState } from 'react'
+import { useResourceQuery } from 'src/hooks/resource'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import Table from 'src/components/Shared/Table'
+import { formatDateDefault } from 'src/lib/date-helper'
+import ErrorWindow from 'src/components/Shared/ErrorWindow'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useWindow } from 'src/windows'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { DataSets } from 'src/resources/DataSets'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import ResourceGlobalForm from './forms/ResourceGlobalForm'
+import AccessLevelForm from './forms/AccessLevelForm'
+import FieldGlobalForm from './forms/FieldGlobalForm'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+
+const GlobalAuthorization = () => {
+  const { getRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [errorMessage, setErrorMessage] = useState(null)
+  const [moduleIdSelection, setModuleIdSelection] = useState(10) //check new way
+
+  async function fetchGridData() {
+    return await getRequest({
+      extension: SystemRepository.ModuleClassRES.qry,
+      parameters: `_filter=&_moduleId=10`
+    })
+  }
+
+  async function fetchWithFilter({ filters }) {
+    return await getRequest({
+      extension: SystemRepository.ModuleClassRES.qry,
+      parameters: `_filter=${filters.qry ?? ''}&_moduleId=${filters.moduleId ?? 10}`
+    })
+  }
+
+  const {
+    query: { data },
+    refetch,
+    labels: labels,
+    filterBy,
+    clearFilter,
+    access,
+    filters
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: SystemRepository.ModuleClassRES.qry,
+    datasetId: ResourceIds.GlobalAuthorization,
+    filter: {
+      endpointId: SystemRepository.ModuleClassRES.qry,
+      filterFn: fetchWithFilter
+    }
+  })
+
+  const onChange = value => {
+    if (value) {
+      filterBy('moduleId', value)
+      setModuleIdSelection(value)
```

Comment:
Why do we need this if it's inside filters?

Reply:
i need to store value of combo selection  inside moduleIdSelection inorder to pass it to the forms.. Do you suggest another way?

Reply:
fixed